Source/ProgressionSystemRuntime/Private/Data/PSWorldSubsystem.cpp

```diff
@@ -122,7 +122,13 @@ void UPSWorldSubsystem::RegisterSpotComponent(UPSSpotComponent* MySpotComponent)
 void UPSWorldSubsystem::OnInitialized_Implementation()
 {
 	StarDynamicProgressMaterial = UMaterialInstanceDynamic::Create(UPSDataAsset::Get().GetDynamicProgressionMaterial(), this);
-	if (!ensureMsgf(StarDynamicProgressMaterial, TEXT("ASSERT: [%i] %hs:\n'StarDynamicProgressMaterial' is null!"), __LINE__, __FUNCTION__))
+	StarLockedProgressMaterial = UMaterialInstanceDynamic::Create(UPSDataAsset::Get().GetDynamicProgressionMaterial(), this);
```

Comment:
Please move Dynamic Material to the local variable above to avoid 3 calls:
```
StarMaterial = UPSDataAsset::Get().GetDynamicProgressionMaterial();
StarDynamic = UMaterialInstanceDynamic::Create(StarMaterial, this);
StarLocked = UMaterialInstanceDynamic::Create(StarMaterial, this);
StarUnlocked = UMaterialInstanceDynamic::Create(StarMaterial, this);
```